### PR TITLE
feat: Support MiniMax-H3 (CORE-375)

### PR DESCRIPTION
**Pull request by kijai**
## Add MiniMax H3 audio-video model support

Support for MiniMax H3, a single-stream packed-token DiT that denoises video (24ch, 16x spatial / 17k+5 frame grid) and stereo audio (32ch, 40 Hz) latents jointly, conditioned on Qwen3-VL-32B hidden states with per-token modality tags.

- DiT with packed sequence layout [text | cond/ref blocks | audio | video], supporting t2va, first/last-frame (fl2va), and reference image/video/audio (ref2va) conditioning. The audio stream runs on its own shifted flow schedule, mapped from the video sigma in closed form so any stock sampler works on the flat AV pack. 
- Supports both original time-embedder and our pruned (40% smaller) precomputed adaln-curve-table checkpoint variants.
- Video VAE: 3D causal CNN encoder + ViT3D decoder with internal spatial tiling and temporal chunking.
- Audio VAE: DAC-lineage encoder + BigVGAN decoder, stereo at 32 kHz (800 samples per latent frame).
- Text encoder: Qwen3-VL-32B truncated to 50 layers, consumed as the unnormalized last hidden state, non-chat-templated presentation with `<Picture i> / <Video k> / <Audio j>` labels and 2 fps timestamped video blocks.

### Nodes: 

EmptyMiniMaxH3LatentAV
MiniMaxH3ImageToVideo
MiniMaxH3ReferenceToVideo
MiniMaxH3SigmaShift

Reuses the LTXV AV latent concat/separate nodes (display names generalized).

### Not model specific changes: 

- ModelOpt AWQ-style pre_quant_scale support in quant ops

- fused activation+quantize path for INT8 linears (linear_input_act).

### CODE REVIEW

File: comfy_extras/nodes_minimax_h3.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    return {"samples": comfy.nested_tensor.NestedTensor((video, audio))}, frame_count
+
+
+class EmptyMiniMaxH3LatentAV(io.ComfyNode):
+    @classmethod
+    def define_schema(cls):
+        return io.Schema(
+            node_id="EmptyMiniMaxH3LatentAV",
+            display_name="Empty MiniMax H3 AV Latent",
+            category="latent/video",
```
Comment (alexisrolland) [inline comfy_extras/nodes_minimax_h3.py]:
```suggestion
            category="model/latent/minimax",
```

File: comfy_extras/nodes_minimax_h3.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+class MiniMaxH3ImageToVideo(io.ComfyNode):
+    """t2va and fl2va: prompt (+ optional first/last keyframes) -> conditioning + AV latent."""
+
+    @classmethod
+    def define_schema(cls):
+        return io.Schema(
+            node_id="MiniMaxH3ImageToVideo",
+            display_name="MiniMax H3 Image to Video",
+            category="conditioning/video_models",
```
Comment (alexisrolland) [inline comfy_extras/nodes_minimax_h3.py]:
```suggestion
            category="model/conditioning/minimax",
```

File: comfy_extras/nodes_minimax_h3.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+    <Picture i> / <Video k> / <Audio j>.
+    """
+
+    @classmethod
+    def define_schema(cls):
+        return io.Schema(
+            node_id="MiniMaxH3ReferenceToVideo",
+            description="<Picture i> / <Video k> / <Audio j> reference conditioning for MiniMax H3. Use the same tags when prompting.",
+            display_name="MiniMax H3 Reference to Video",
+            category="conditioning/video_models",
```
Comment (alexisrolland) [inline comfy_extras/nodes_minimax_h3.py]:
```suggestion
            category="model/conditioning/minimax",
```

File: comfy/ldm/minimax/audio_vae.py [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+        self.q_bias = nn.Parameter(torch.empty(in_dim))
+        self.v_bias = nn.Parameter(torch.empty(in_dim))
+        self.register_buffer("zero_k_bias", torch.empty(in_dim))
+        self.proj = ops.Linear(out_dim, out_dim)
+
+    def forward(self, x):
+        B, N, C = x.shape
+        weight, _, offload_stream = comfy.ops.cast_bias_weight(self.qkv, x, offloadable=True)
+        qkv = F.linear(x, weight=weight, bias=torch.cat((self.q_bias, self.zero_k_bias, self.v_bias)))
+        comfy.ops.uncast_bias_weight(self.qkv, weight, None, offload_stream)
```
[the remark this answers is not in the document]
Comment (kijai) [reply comfy/ldm/minimax/audio_vae.py:243]: If published weights omit pre_block.attn.zero_k_bias - they won't

File: comfy_extras/nodes_minimax_h3.py [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+    and derives the audio schedule from it.
+    """
+
+    @classmethod
+    def define_schema(cls):
+        return io.Schema(
+            node_id="MiniMaxH3SigmaShift",
+            description="Set the video/audio flow shifts.",
+            display_name="MiniMax H3 Sigma Shift",
+            category="advanced/model",
```
Comment (alexisrolland) [inline comfy_extras/nodes_minimax_h3.py]:
```suggestion
            category="model/patch/minimax",
```

File: comfy/ldm/minimax/audio_vae.py [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+        right_pad = math.ceil(length / self.hop_length) * self.hop_length - length
+        waveform = F.pad(waveform, (0, right_pad))
+        x = waveform.reshape(b * s, 1, -1)
+        x = self.encoder(x)  # [b * s, latent_dim, T]
+        x = self.pre_block(x.transpose(1, 2)).transpose(1, 2)  # [b * s, 32, T]
+        z = self.mean_proj(x)
+        mean = self.latents_mean.view(1, -1, 1).to(device=z.device, dtype=z.dtype)
+        std = self.latents_std.view(1, -1, 1).to(device=z.device, dtype=z.dtype)
+        z = (z - mean) / std
+        return z.reshape(b, s, z.shape[1], z.shape[2]).permute(0, 2, 1, 3)
```
[the remark this answers is not in the document]
Comment (kijai) [reply comfy/ldm/minimax/audio_vae.py:442]: Not true.

File: comfy/ldm/minimax/model.py [hunk 7]
```diff
@@ -0,0 +1,659 @@
...
+        aug = payload.get("audio_cond_noise_aug", AUDIO_COND_TIMESTEP)
+        seed = int(payload.get("seed", 0)) + 1
+        for z in payload.get("cond_audio_latents", []):
+            r = pack_audio(z.to(torch.float32))
+            if aug < 1.0:
+                gen = torch.Generator("cpu").manual_seed(seed)
+                noise = torch.randn(r.shape, generator=gen, dtype=torch.float32)
+                r = aug * r + (1.0 - aug) * noise.to(r.device)
+            rows.append(r.to(device))
+        return torch.cat(rows, dim=0) if rows else None
```
[the remark this answers is not in the document]
Comment (kijai) [reply comfy/ldm/minimax/model.py:496]: Surely this is too trivial to optimize?

**Review by alexisrolland — APPROVED** [review]

**Review by comfyanonymous — APPROVED** [review]
